Add notifications and feeds

samchuk-vlad opened the pull request.

samchuk-vlad committed:
WIP: add notifications and feeds

samchuk-vlad committed:
WIP: refactor code

- Remove comments

samchuk-vlad committed:
WIP: add websocket and push notification and feeds

samchuk-vlad committed:
WIP: add profile

- Refactor code

samchuk-vlad committed:
WIP: add settings

- Refactor code

olehmell commented on .env:
```diff
@@ -0,0 +1,14 @@
+
+
+IPFS_URL=http://127.0.0.1:8080
+# https://app.subsocial.network/ipfs
```
Remove `.env`

olehmell commented on src/Feed/Feed.ts:
```diff
@@ -0,0 +1,62 @@
+export const getPostPreview = async (feed: Activity): Promise<string> => {
+	const subsocial = await resolveSubsocialApi()
+	const { date, post_id, account, space_id } = feed
+	const formatDate = new Date(date).toUTCString()
```
Use `dayjs`

olehmell commented on src/Notifications/Notifications.ts:
```diff
@@ -0,0 +1,128 @@
+	return createMessageForNotifs(formatDate, accountUrl, msg, followingUrl)
+}
+
+const getSpacePreviewWithMaps = async (account: string, spaceId: string, msg: string, date: string): Promise<string> => {
```
Why `withMaps` maybe delete?

olehmell commented on src/Notifications/Notifications.ts:
```diff
@@ -0,0 +1,128 @@
+	return createMessageForNotifs(formatDate, createHrefForAccount(account, accountName), msg, url)
+}
+
+export const showNotification = async (ctx: TelegrafContext, notifOffset: number) => {
```
`ctx: { reply, telegram: { sendMessage } }`

olehmell commented on src/Profile/Profile.ts:
```diff
@@ -0,0 +1,36 @@
+	const profile = await subsocial.findProfile(account)
+	if (!profile) return
+
+	const accountName = profile.content?.name ? profile.content.name : ""
```
```suggestion
	const accountName = profile.content?.name ||  ''
```

olehmell commented on src/Profile/Profile.ts:
```diff
@@ -0,0 +1,36 @@
+		accountName,
+		account,
+		freeBalance,
+		reputation.toString(),
```
Dont use `toString`

olehmell commented on src/Settings/settings.ts:
```diff
@@ -0,0 +1,25 @@
+}
+
+export const showSettings = async (ctx: TelegrafContext) => {
+	const account = await getAccountByChatId(ctx.chat.id)
```
`const charId = ctx.chat.id`

olehmell commented on src/index.ts:
```diff
@@ -1,53 +1,80 @@
+bot.use(stage.middleware())
+
+bot.start(async (ctx) => {
+  await ctx.telegram.sendMessage(ctx.chat.id, 'Hi in subsocial telegram bot')
```
```suggestion
  await ctx.telegram.sendMessage(ctx.chat.id, 'Hi in Subsocial telegram bot')
```

olehmell commented on src/scenes.ts:
```diff
@@ -0,0 +1,30 @@
+		})
+		scene.on('text', async (ctx) => {
+			const message = ctx.message.text
+			const isValidAccount = checkAddress(message, 28)
```
```suggestion
			const isValidAccount = !!checkAddress(message, 28)[0]
```

olehmell commented on src/scenes.ts:
```diff
@@ -0,0 +1,30 @@
+		scene.on('text', async (ctx) => {
+			const message = ctx.message.text
+			const isValidAccount = checkAddress(message, 28)
+			if (isValidAccount[0]) {
```
```suggestion
			if (isValidAccount) {
```

olehmell commented on src/scenes.ts:
```diff
@@ -0,0 +1,30 @@
+import { mainMenuKeyboard } from './index';
+import { setTelegramData, getTelegramChat } from './utils/OffchainUtils';
+import { checkAddress } from '@polkadot/util-crypto';
+const Scene = require('telegraf/scenes/base')
```
Why dont use `import`?

olehmell commented on src/utils/OffchainUtils.ts:
```diff
@@ -0,0 +1,98 @@
+import axios from 'axios'
+import { newLogger } from '@subsocial/utils'
+import { Activity as OldActivity } from '@subsocial/types'
```
```suggestion
import { Activity } from '@subsocial/types'
```

olehmell commented on src/utils/OffchainUtils.ts:
```diff
@@ -0,0 +1,98 @@
+import { offchainUrl } from '../env'
+require('dotenv').config()
+
+export type Activity = Omit<OldActivity, 'id'> & {
```
Remove it

olehmell commented on src/utils/utils.ts:
```diff
@@ -0,0 +1,80 @@
+import { updateTelegramChat, getTelegramChat, getAccountByChatId } from './OffchainUtils';
+
+export type Type = 'notification' | 'feed'
+
```
Maybe reuse similar functions from UI?

Folder: `components/urls/helpers.ts`

olehmell commented on src/Notifications/Notifications.ts:
```diff
@@ -0,0 +1,128 @@
+	Markup.callbackButton('Load more', 'loadMoreNotifs')
+  ])
+
+export const createNotificationMessage = async (activities: Activity[]) => {
```
```suggestion
export const createNotificationsMessage = async (activities: Activity[]) => {
```

olehmell commented on src/ws.ts:
```diff
@@ -0,0 +1,49 @@
+	const { push_notifs, push_feeds } = await getTelegramChat(account, chatId)
+
+	if (type === 'notification' && push_notifs) {
+		const notifMessage = await createNotificationMessage([activity])
```
create function for one activity 
```createNotificationMessage = (activity) => getFirstOrUndefined(createNotificationsMessage([ activity ])```

olehmell commented on src/Feed/Feed.ts:
```diff
@@ -0,0 +1,62 @@
+import { PostId, SpaceId } from '@subsocial/types/substrate/interfaces';
+import { resolveSubsocialApi } from '../Substrate/subsocialConnect';
+import {
+	createHrefForPost,
```
Change tab size from 4 to 2

samchuk-vlad committed:
Refactor code

- add switch account and sign out
- add current user

samchuk-vlad committed:
Add space in account

samchuk-vlad committed:
Add last push blockNumber and eventIndex

- Catch error with delete from group chat
- Add auto-leave from group chat
- Fix bug with empty profiles

samchuk-vlad committed:
Rename method name

samchuk-vlad committed:
Add icons for account and settings

F3Joule committed:
Update project files

F3Joule committed:
Refactor file structure

- Fix dotenv

samchuk-vlad committed:
Refactor path in OffchainUtils

samchuk-vlad committed:
Merge remote-tracking branch 'F3Joule/master' into vs-notifs-and-reeds

olehmell committed:
Update to new version packages

olehmell committed:
Comment out repitation of profile

olehmell committed:
Bump version

olehmell committed:
Fix typo

F3Joule committed:
Update .gitignore

F3Joule committed:
Fix English a little

F3Joule committed:
Remove redundant ipfs.ts

F3Joule committed:
Move subsocialConnect to utils

F3Joule committed:
Fix indents and reformat code

samchuk-vlad committed:
Merge pull request #1 from F3Joule/vs-notifs-and-reeds

Fix code formatting, indents and English typos in text messages
